fast64_internal/oot/oot_cutscene.py

```diff
@@ -0,0 +1,604 @@
+	box = layout.column(align=True)
+	l = 0
+	row = box.row(align=True)
+	row.label(text = 'Add:')
+	addButton(row)
+	for _ in range(3):
+		row = box.row(align=True)
+		for _ in range(3):
+			addButton(row)
+	box.label(text = 'Install zcamedit for camera/actor motion.')
+
+
+class OOTCutsceneProperty(bpy.types.PropertyGroup):
+	csEndFrame : bpy.props.IntProperty(name = "End Frame", min = 0, default = 100)
+	csWriteTerminator : bpy.props.BoolProperty(name = "Write Terminator (Code Execution)")
+	csTermIdx : bpy.props.IntProperty(name = "Index", min = 0)
+	csTermStart : bpy.props.IntProperty(name = "Start Frm", min = 0, default = 99)
+	csTermEnd : bpy.props.IntProperty(name = "End Frm", min = 0, default = 100)
+	csLists : bpy.props.CollectionProperty(type = OOTCSListProperty, name = 'Cutscene Lists')
+
+def drawCutsceneProperty(box, obj):
+	prop = obj.ootCutsceneProperty
+	box.prop(prop, "csEndFrame", text = "End Frame")
+	box.prop(prop, "csWriteTerminator", text = "Write Terminator (Code Execution)")
+	if prop.csWriteTerminator:
+		r = box.row()
+		r.prop(prop, "csTermIdx", text = "Index")
+		r.prop(prop, "csTermStart", text = "Start Frm")
+		r.prop(prop, "csTermEnd", text = "End Frm")
+	for i, p in enumerate(prop.csLists):
+		drawCSListProperty(box, p, i, obj.name, 'Cutscene')
+	drawCSAddButtons(box, obj.name, 'Cutscene')
+
+
+################################################################################
+# Properties to level classes
+################################################################################
+
+def readCutsceneData(csParentOut, csParentIn):
+	for listIn in csParentIn.csLists:
+		listOut = OOTCSList()
+		listOut.listType = listIn.listType
+		listOut.unkType, listOut.fxType, listOut.fxStartFrame, listOut.fxEndFrame \
+		 	= listIn.unkType, listIn.fxType, listIn.fxStartFrame, listIn.fxEndFrame
+		listData = []
+		if listOut.listType == 'Textbox':
+			for entryIn in listIn.textbox:
+				entryOut = OOTCSTextbox()
+				entryOut.textboxType = entryIn.textboxType
+				entryOut.messageId = entryIn.messageId
+				entryOut.ocarinaSongAction = entryIn.ocarinaSongAction
+				entryOut.startFrame = entryIn.startFrame
+				entryOut.endFrame = entryIn.endFrame
+				entryOut.type = entryIn.type
+				entryOut.topOptionBranch = entryIn.topOptionBranch
+				entryOut.bottomOptionBranch = entryIn.bottomOptionBranch
+				entryOut.ocarinaMessageId = entryIn.ocarinaMessageId
+				listOut.entries.append(entryOut)
+		elif listOut.listType == 'Lighting':
+			for entryIn in listIn.lighting:
+				entryOut = OOTCSLighting()
+				entryOut.index = entryIn.index
+				entryOut.startFrame = entryIn.startFrame
+				listOut.entries.append(entryOut)
+		elif listOut.listType == 'Time':
+			for entryIn in listIn.time:
+				entryOut = OOTCSTime()
+				entryOut.startFrame = entryIn.startFrame
+				entryOut.hour = entryIn.hour
+				entryOut.minute = entryIn.minute
+				listOut.entries.append(entryOut)
+		elif listOut.listType in ['PlayBGM', 'StopBGM', 'FadeBGM']:
```

Comment:
```suggestion
		elif listOut.listType in {'PlayBGM', 'StopBGM', 'FadeBGM'}:
```
speeeeed

Reply:
That is extremely pedantic but I suppose strictly speaking superior. Changed.

Reply:
It's 3 times faster! Speeeed